File: Service-Hub/ServiceHub.DAL/Entities/Order.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using ServiceHub.DAL.Enums;
using ServiceHub.DAL.Helper;

namespace ServiceHub.DAL.Entities
{
    public class Order
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public int UserId { get; set; }
        public int WorkerId { get; set; }
        [Required]
        public OrderStatus Status { get; set; } = OrderStatus.New;
        public DateTime CreatedDate { get; set; } = DateTime.Now;

        [ForeignKey("UserId")]
        public ApplicationUser User { get; set; }
        [ForeignKey("WorkerId")]
        public ApplicationUser Worker { get; set; }

        public Order()
        {
            //Status = OrderStatus.New;
            CreatedDate = DateTime.Now;
        }


    }
}

File: Service-Hub-Frontend/src/app/Models/order.ts
import { OrderStatus } from "../enums/order-status";
import { Worker } from "./worker.model";

export class Order {
    constructor(
        public id:number,
        // public userId:number,
        // public workerId:number,
        public user:Worker,
        public worker:Worker,
        public status: OrderStatus,
        public createdDate: Date
    ){}
}
